node/node.go

```diff
@@ -235,7 +242,7 @@ func (n *Node) Reconfigure(ctx context.Context, deps resource.Dependencies, conf
 }
 
 // getGateway sends the validate docommand to the gateway to confirm the dependency.
-func getGateway(ctx context.Context, deps resource.Dependencies) (sensor.Sensor, error) {
+func (n *Node) getGateway(ctx context.Context, deps resource.Dependencies) (sensor.Sensor, error) {
```

Comment:
either we should return both the region and the gateway, or store them both on the struct within this function